lib/rubygems/gemcutter_utilities.rb

```diff
@@ -132,12 +133,26 @@ def sign_in(sign_in_host = nil)
 
     email = ask "   Email: "
     password = ask_for_password "Password: "
+
+    hostname = Socket.gethostname
+    default_key_name = "#{hostname}"
+    key_name = ask "Key name [#{default_key_name}]: "
+    key_name = default_key_name if key_name.empty?
+
+    scope_params = {}
+
+    say "Please select scopes you want to enable for the key (y/n)"
+    API_SCOPES.each do |scope|
+      selected = ask "#{scope} [n]: "
+      scope_params[scope] = true if selected == "y"
```

Comment:
You will need to cover other different variants, ie: `y|Y|yes` etc. IIRC rubygems handles this already in some places.

Reply:
I have updated this to a regex.